### PR TITLE
elixir: force recompile with erlang 25

### PR DESCRIPTION
**Pull request by robsonpeixoto**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by iMichka** [comment]
Thanks. Do you think there is a way to improve our test to catch this case earlier?
It looks like `system "#{bin}/elixir", "-v"` is not enough to detect the need for a revision bump.

**Comment by robsonpeixoto** [comment]
@iMichka , the command `elixir -v` in enough to get it:
```
Elixir 1.13.4 (compiled with Erlang/OTP 24)
```

The test is ignoring the output.

**Comment by iMichka** [comment]
Could you try to update the test and assert_match for `(compiled with Erlang/OTP 25)`, thus forcing an update on our side next time we make an Erlang update?

**Comment by robsonpeixoto** [comment]
Done in https://github.com/Homebrew/homebrew-core/pull/108141/commits/bc964f7dfaf46ae6ddbe70def9221dd9f0e4311a

**Comment by iMichka** [comment]
> Use `%r` around regular expression.

Can you try to fix this? Should be good to be merged once it's done.

**Comment by robsonpeixoto** [comment]
Done in 8fe5c0edbdb456e104fcd742c70033656825c498

**Review by iMichka — APPROVED** [review]

**Comment by iMichka** [comment]
Thanks!

File: Formula/elixir.rb [hunk 1]
```diff
@@ -30,6 +31,6 @@ def install
   end
 
   test do
-    system "#{bin}/elixir", "-v"
+    assert_match(%r{(compiled with Erlang/OTP 25)}, shell_output("#{bin}/elixir -v"))
```
Comment (chenrui333) [inline Formula/elixir.rb:34]: Coming from #116834, I think this hardcoded value is not a good test. erlang dependency would be changed over time.

**Pull request closed without merging**